state/state.go

```diff
@@ -0,0 +1,90 @@
+package state
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/md5"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+)
+
+// EncryptState - Creates an encrypted state for later verification
+func EncryptState(clientID string, key string) string {
+	hashedKey := createHash(addSalt(key))
+	encryptedState, err := encrypt([]byte(clientID), []byte(hashedKey))
+	if err != nil {
+		fmt.Println(err)
+		return ""
+	}
+	return base64.StdEncoding.EncodeToString(encryptedState)
+}
+
+// DecryptState - Decrypts the given state and returns the clientID or an error
+func DecryptState(state string, key string) (string, error) {
+	encryptedState, err := base64.StdEncoding.DecodeString(state)
+	if err != nil {
+		return "", err
+	}
+
+	hashedKey := createHash(addSalt(key))
+	clientID, err := decrypt([]byte(encryptedState), []byte(hashedKey))
+	if err != nil {
+		return "", err
+	}
+	return string(clientID), nil
+}
+
+func addSalt(key string) string {
+	salt := time.Now().Format("2006-01-02")
```

Comment:
Okay, this is not really the best salt I could add, but this will only be used for myself so it's fine for me